src/Spectre.libStatistics/Math.h

```diff
@@ -380,4 +380,30 @@ std::vector<DataType> build(size_t size, Generator generator)
     return result;
 }
 
+/// <summary>
+/// Calculates differences between adjacent elements of given container.
+/// </summary>
+/// <param name="data">The container.</param>
+/// <param name="order">Order of difference operation (how many times the difference is executed).</param>
+/// <returns>Vector of calculated differences.</returns>
+ template <class DataType>
+ std::vector<DataType> diff(gsl::span<const DataType> data, uint16_t order = 1)
```

Comment:
`std::vector<std::remove_const<DataType>::type> diff(gsl::span<DataType> data, uint16_t order = 1)`

Reply:
What? Why would you want to change the signature of the function to return type that differs from every single return type of any mathematical function in here? Please elaborate.